client/src/pages/AddProject.js

```diff
@@ -0,0 +1,227 @@
+import validator from 'validator';
+
+import { CustomOutlinedInput, OutlinedSelect, handleSelect } from '../components/Inputs';
+import UploadImages from '../components/UploadImages';
+import FormValidator from '../helpers/form-validation';
+import { withToast } from '../components/Toast';
+import { addProject } from '../api/projects';
+
+const styles = {
+    pageContent: {
+        display: 'flex',
+        justifyContent: 'center'
+    },
+    projectPreview: {
+        margin: "10px"
+    },
+    addProjectPage: {
+        margin: "10px"
+    },
+    form: {
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'left',
+        width: '100%'
+    },
+}
+
+const industries = [
+    { id: 0, name: '' },
+    { id: 1321, name: 'Technology' },
+    { id: 21423, name: 'Marketing' },
+    { id: 42342, name: 'Engineering' },
+    { id: 21342, name: 'Art' },
+    { id: 93082, name: 'Film' }
+]
+
+const locations = [
+    { id: 0, name: '' },
+    { id: 123, name: 'New York' },
+    { id: 431, name: 'California' },
+    { id: 903, name: 'Georgia' },
+    { id: 223, name: 'Florida' }
+]
+
+class AddProject extends Component {
+    constructor(props) {
+        super(props);
+        this.validators = FormValidator([
+            {
+                field: 'title',
+                method: validator.isEmpty,
+                validWhen: false,
+                message: 'Title is required.'
+            },
+            {
+                field: 'industry',
+                method: validator.isEmpty,
+                validWhen: false,
+                message: 'Industry is required'
+            },
+            {
+                field: 'location',
+                method: validator.isEmpty,
+                validWhen: false,
+                message: 'location is required'
+            },
+            {
+                field: 'fundingGoal',
+                method: validator.isNumeric,
+                validWhen: true,
+            }
+        ]);
+        this.state = {
+            uploading: false,
+            title: '',
+            subtitle: '',
+            industry: '',
+            location: '',
+            images: [],
+            fundingGoal: 0,
+            formData: {},
+            validation: this.validators.valid()
+        }
+    }
+
+    componentDidMount() {
+        this.setState({ formData: new FormData() })
+    }
+
+    handleInput = (event) => {
+        const { value, name } = event.target;
+        const { formData } = this.state;
+        formData.set(name, value);
+        this.setState({ [name]: value, formData })
+    }
+
+    handleSubmit = async (event) => {
+        event.preventDefault();
+        const validation = this.validators.validate(this.state);
+        this.setState({ validation });
+
+        if (validation.isValid) {
+            const { formData } = this.state;
+            const { id } = this.props.match.params; // will have to change eventually.
```

Comment:
Not anything wrong with this line, but bookmarking it for change in later ticket/PR? Maybe: #42 if merged before

Reply:
fyi, could get this from props.userDetails object now that the PR has been merged